flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/extraction/FunctionMappingExtractor.java

```diff
@@ -437,9 +440,30 @@ static ResultExtraction createReturnTypeResultExtraction() {
 
 	/**
 	 * Extraction that uses a generic type variable for producing a {@link FunctionResultTemplate}.
+	 *
+	 * <p>If enabled, a {@link DataTypeHint} from method or class has higher priority.
 	 */
-	static ResultExtraction createGenericResultExtraction(Class<? extends UserDefinedFunction> baseClass, int genericPos) {
+	static ResultExtraction createGenericResultExtraction(
+			Class<? extends UserDefinedFunction> baseClass,
+			int genericPos,
+			boolean allowDataTypeHint) {
```

Comment:
Why do we need the switch (`allowDataTypeHint`) here? Is it only to avoid rewriting tests?

Reply:
it is set to `false` for accumulator generic extraction like here
https://github.com/apache/flink/pull/13149/files#diff-7333ebb65ea0b690c1bf857945f8dcd8R99

we only allow this shortcut for output types, not accumulator types